### DISCUSSION TITLE
Alerts export - is this possible for non-Incident Alerts?

### DISCUSSION THREAD

**Discussion by khntst** [martindstone/pagerduty-cli#39] [Q&A] [answered]
Fantastic tool! Makes some bulk actions feasible without coding, also a more approachable on-ramp when combined with wrapper code.

I was combing through the docs.  Seems like pd-cli can only export Incidents, or alerts associated with Incidents.
Am I correct in this?   To export raw alerts would require an enhancement to pd-cli?  TBH I'm not sure it is even possible to get raw from the API.  But Alerts can exist without Incident relationship when suppressed.

My goal is to do analytics across many/all of alert sources coming into our PagerDuty domain.

**Comment by martindstone** [comment, marked as answer]
Hi @khntst - my apologies for the delay. There's no officially supported endpoint to fetch alerts independent of incidents, but I can tell you that inspecting the PD web UI will reveal that there is a `/alerts` endpoint currently implemented that implements arguments and pagination that's very similar to /incidents... and there's also a `pd rest fetch` command that can fetch endpoints and output them as tables using `-t` and `-k`... and can filter the rows and output CSV just like the other `list` commands... hope this helps!!

**Reply by khntst** [reply]
Thanks so much!
Yours was a huge clue, never dawned on me.  Sure enough, using /alerts as API endpoint in Postman, I was indeed able to (also) retrieve Alerts not linked to an Incident.  In such case, "incident": null.  I proceeded to use pd rest fetch - alerts with pd-cli.  Yup, a good way to get bulk export that I was looking for.  Eliminates the need to write code to iterate through JSON responses and flatten subsets of fields into table.  I can just slurp this into Excel or whatever and slice & dice.

I expect various filter parameters will work also.  Looks like 'since' and 'until' are applied to the REST request itself (not just post-collect filter) -- these were most important.
Probably little risk this endpoint would ever go away, as it is fairly fundamental for the UI.  Strange that they don't document it.

**Reply by martindstone** [reply]
Yeah it's strange for sure... I can confirm that you can pass iso8601 timestamps in since= and until= parameters (the pd rest fetch flag for url params is -P and there's a pd util timestamp command that can translate things like "1 day ago" to an iso8601 ts for use with fetch)
